packages/cli/src/generators/index.js

```diff
@@ -0,0 +1,127 @@
+const path = require('path');
+
+const prettier = require('gulp-prettier');
+
+const {
+  NODE_VERSION,
+  PACKAGE_VERSION,
+  PLATFORM_PACKAGE
+} = require('../constants');
+
+const Generator = require('yeoman-generator');
+
+const AUTH_TYPE_CHOICES = [
+  {
+    name: 'Basic',
+    value: 'basic'
+  },
+  {
+    name: 'Custom (API Key)',
+    value: 'custom'
+  },
+  {
+    name: 'Digest',
+    value: 'digest'
+  },
+  {
+    name: 'OAuth1',
+    value: 'oauth1'
+  },
+  {
+    name: 'OAuth2',
+    value: 'oauth2'
+  },
+  {
+    name: 'Session',
+    value: 'session'
+  }
+];
+
+class ProjectGenerator extends Generator {
+  initializing() {
+    this.sourceRoot(path.resolve(__dirname, 'templates'));
+    this.destinationRoot(path.resolve(this.options.path));
+
+    this.registerTransformStream(prettier({ singleQuote: true }));
+  }
+
+  async prompting() {
+    if (!this.options.authType) {
+      this.answers = await this.prompt([
+        {
+          type: 'list',
+          name: 'authType',
+          choices: AUTH_TYPE_CHOICES,
+          message: "What's your authentication type?"
+        }
+      ]);
+      this.options.authType = this.answers.authType;
+    }
+  }
+
+  _writeReadme() {
+    this.fs.copyTpl(
+      this.templatePath('README.template.md'),
+      this.destinationPath('README.md'),
+      { name: this.options.packageName }
+    );
+  }
+
+  _writePackageJson() {
+    this.fs.writeJSON('package.json', {
+      name: this.options.packageName,
+      version: '1.0.0',
+      description: '',
+      main: 'index.js',
+      scripts: {
+        test: 'mocha --recursive -t 10000 test'
+      },
+      engines: {
+        node: `>=${NODE_VERSION}`,
+        npm: '>=5.6.0'
+      },
+      dependencies: {
+        [PLATFORM_PACKAGE]: PACKAGE_VERSION
+      },
+      devDependencies: {
+        mocha: '^5.2.0',
+        should: '^13.2.0'
```

Comment:
I think i'm going to try to swap at least `should` in the new templates - it's also been a long-held frustration with me. I find it very backwards. 

`jest` has `expect` included so we could use that as a runner. I've been super impressed with the guard rails and suggestions they provide (for instance, suggesting to use different or better assertion functions when you'll get unexpected behavior). I think we could also use that separate though if you want.

Reply:
Yeah, I'm not a fan of `should` either. I'll give `jest` a test today.